src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentOperationService.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// provide various operations for this document
+    /// 
+    /// I followed name from EditorOperation for now. 
+    /// </summary>
+    internal interface IDocumentOperationService : IDocumentService
+    {
+        /// <summary>
+        /// document version of <see cref="Workspace.CanApplyChange(ApplyChangesKind)"/>
+        /// </summary>
+        bool CanApplyChange { get; }
```

Comment:
should these maybe be methods?

Reply:
not right now, but it might end up become method. but for now, it is property.